File: backend-ms1/src/main/java/ma/zs/easystock/service/facade/admin/gestionchequeffet/TransfertAdminService.java
package ma.zs.easystock.service.facade.admin.gestionchequeffet;

import java.util.List;
import ma.zs.easystock.bean.core.gestionchequeffet.Transfert;
import ma.zs.easystock.dao.criteria.core.gestionchequeffet.TransfertCriteria;
import ma.zs.easystock.zynerator.service.IService;



public interface TransfertAdminService extends  IService<Transfert,TransfertCriteria>  {

    List<Transfert> findByRaisonId(Long id);
    int deleteByRaisonId(Long id);
    long countByRaisonId(Long id);



}

File: frontend/src/app/controller/service/admin/gestionchequeffet/TransfertAdmin.service.ts
import { Injectable } from '@angular/core';
import {HttpClient} from '@angular/common/http';
import {BehaviorSubject, Observable} from 'rxjs';

import { RoleService } from 'src/app/zynerator/security/Role.service';
import {environment} from 'src/environments/environment';

import {TransfertDto} from 'src/app/controller/model/gestionchequeffet/Transfert.model';
import {TransfertCriteria} from 'src/app/controller/criteria/gestionchequeffet/TransfertCriteria.model';
import {AbstractService} from 'src/app/zynerator/service/AbstractService';


@Injectable({
  providedIn: 'root'
})
export class TransfertAdminService extends AbstractService<TransfertDto, TransfertCriteria> {
     constructor(private http: HttpClient, private roleService: RoleService) {
        super();
        this.setHttp(http);
        this.setApi(environment.apiUrl + 'admin/transfert/');
    }

    public constrcutDto(): TransfertDto {
        return new TransfertDto();
    }

    public constrcutCriteria(): TransfertCriteria {
        return new TransfertCriteria();
    }
}
